crates/mun_lld/build.rs

```diff
@@ -0,0 +1,415 @@
+    // there
+    let s = match c.name("patch") {
+        None => format!("{}.0", &c[0]),
+        Some(_) => c[0].to_string(),
+    };
+    Ok(Version::parse(&s).unwrap())
+}
+
+/// Get the names of the dylibs required by LLVM, including the C++ standard
+/// library.
+fn get_system_libraries() -> Vec<String> {
+    llvm_config("--system-libs")
+        .split(&[' ', '\n'] as &[char])
+        .filter(|s| !s.is_empty())
+        .map(|flag| {
+            if cfg!(target_env = "msvc") {
+                // Same as --libnames, foo.lib
+                assert!(flag.ends_with(".lib"));
+                &flag[..flag.len() - 4]
+            } else {
+                // Linker flags style, -lfoo
+                assert!(flag.starts_with("-l"));
+                &flag[2..]
+            }
+        })
+        .chain(get_system_libcpp())
+        .map(str::to_owned)
+        .collect::<Vec<String>>()
+}
+
+/// Get the library that must be linked for C++, if any.
+fn get_system_libcpp() -> Option<&'static str> {
+    if cfg!(target_env = "msvc") {
+        // MSVC doesn't need an explicit one.
+        None
+    } else if cfg!(target_os = "macos") {
+        // On OS X 10.9 and later, LLVM's libc++ is the default. On earlier
+        // releases GCC's libstdc++ is default. Unfortunately we can't
+        // reasonably detect which one we need (on older ones libc++ is
+        // available and can be selected with -stdlib=lib++), so assume the
+        // latest, at the cost of breaking the build on older OS releases
+        // when LLVM was built against libstdc++.
+        Some("c++")
+    } else if cfg!(target_os = "freebsd") {
+        Some("c++")
+    } else {
+        // Otherwise assume GCC's libstdc++.
+        // This assumption is probably wrong on some platforms, but would need
+        // testing on them.
+        Some("stdc++")
+    }
+}
+
+/// Get the names of libraries to link against.
+fn get_link_libraries() -> Vec<String> {
+    // Using --libnames in conjunction with --libdir is particularly important
+    // for MSVC when LLVM is in a path with spaces, but it is generally less of
+    // a hack than parsing linker flags output from --libs and --ldflags.
+    llvm_config("--libnames")
+        .split(&[' ', '\n'] as &[char])
+        .filter(|s| !s.is_empty())
+        .map(|name| {
+            // --libnames gives library filenames. Extract only the name that
+            // we need to pass to the linker.
+            if cfg!(target_env = "msvc") {
+                // LLVMfoo.lib
+                assert!(name.ends_with(".lib"));
+                &name[..name.len() - 4]
+            } else {
+                // libLLVMfoo.a
+                assert!(name.starts_with("lib") && name.ends_with(".a"));
+                &name[3..name.len() - 2]
+            }
+        })
+        .map(str::to_owned)
+        .collect::<Vec<String>>()
+}
+
+// fn get_llvm_cflags() -> String {
```

Comment:
If we don't need this, it can be deleted :)

Reply:
I removed it